### PR TITLE
refactor(cloudflare): rename the "cloudflare" wrapper to "cloudflare-edge"

### PR DESCRIPTION
**Pull request by vicb**
The commit message do not mention "cloudflare-streaming" -> "cloudflare-node" as this was never released officially.

### CODE REVIEW

File: packages/open-next/src/build/helper.ts [hunk 1]
```diff
@@ -356,3 +359,18 @@ export function initOutputDir(options: BuildOptions) {
...
+ * @returns Whether the edge runtime is used
+ */
+export async function isEdgeRuntime(
+  overrides: DefaultOverrideOptions | undefined,
+) {
+  if (!overrides?.wrapper) {
+    return false;
+  }
+  if (typeof overrides.wrapper === "string") {
+    return overrides.wrapper.startsWith("cloudflare");
```
Comment (conico974) [inline packages/open-next/src/build/helper.ts:373]: This is weird, `cloudflare-node` does run in an `edge` like runtime but is meant for the `node` runtime of next.

**Review by conico974 — COMMENTED** [review]
I think we should change the `Wrapper` type so that it is clearer (and would allow for better validation). It could become 
```ts
export type Wrapper<
  E extends BaseEventOrResult = InternalEvent,
  R extends BaseEventOrResult = InternalResult,
> = BaseOverride & {
  wrapper: WrapperHandler<E, R>;
  supportStreaming: boolean;
  supportExternalMiddleware?: boolean
  edgeLikeRuntime?: boolean;
};
```

`edgeLikeRuntime` would be used at build time to apply the necessary fix for cloudflare. It would also help distinguish this from Next `edge` runtime. This would be set for both cloudflare wrappers.
I don't like this name a lot, maybe you can find something better. Maybe something with Worker or WinterCG 

`supportExternalMiddleware` would only be used for validation because external middleware need to be able to handle both `InternalResult` and `MiddlewareOutputEvent`. This one would be set to true for the `aws-cloudfront` wrapper as well.

All of this could be done in a different PR if you prefer.

**Comment by vicb** [comment]
> All of this could be done in a different PR if you prefer.

Yes, I would prefer a different PR as I think we probably need to refactor the code.
See https://github.com/opennextjs/opennextjs-aws/issues/646

File: packages/open-next/src/build/helper.ts (hunk 1 above)
Comment (vicb) [reply packages/open-next/src/build/helper.ts]: fixed

**Review by conico974 — APPROVED** [review]
LGTM Thanks

**Comment by vicb** [comment]
Thanks for the review!

**Pull request merged**